update

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+set -ex
+
+BRANCH=${1:-master}
+
+git clean -f -d
+GENERATED_PATHS=$(ls -1)
+
+# checkout source branch
+git checkout "$BRANCH" && git submodule update --init
+COPY_PATHS=$(ls -1)
+tag=$(git rev-parse HEAD)
+
+# copy the branch content into source/
+mkdir source
+for i in $COPY_PATHS; do
+    cp -r "$i" source
+done
+VERSIONS=$(sed -n 's/^VERSIONS[[:space:]]*=//p' source/Makefile)
+
+git checkout generated
+# Sometimes git does not remove common/ on branch change
+[ -d common ] && rm -rf common
+
+for i in $GENERATED_PATHS; do
+    case "$i" in
+    # Do not remove the update script itself
+    update)
+        continue
+        ;;
+    *)
+        rm -rf "$i"
```

Comment:
I would pay a lot more attention to this, `$GENERATED_PATHS` is an output from `ls -1`, which might well contain some spaces in dirnames, and those could lead to accidental removes, maybe that's not huge security issue...
Maybe IFS='\n' is needed.

Reply:
True, thanks for catching this. Will fix in next version.